userspace/libsinsp/sinsp_filtercheck_thread.cpp

```diff
@@ -1783,6 +1972,13 @@ bool sinsp_filter_check_thread::compare(sinsp_evt *evt)
 			return compare_full_acmdline(evt);
 		}
 	}
+	else if(m_field_id == TYPE_AENV)
```

Comment:
AFAIK this flow kicks in when you use `proc.aenv` and the other as filter expression in your rules when you search for `proc.aenv contains xyz`. That being said, we can always check in the next dev cycle if things could be refactored.

Reply:
uhm got it, so `proc.aenv` without arguments will have 2 different behaviors:
* if used as an output, it will render the `proc.env`
* if used in a condition it will allow to check the presence of an env, among all ancestors

if this is true, I think your initial description of the field was right

Reply:
>  I think your initial description of the field was more precise

reverted my commit!